package/MDAnalysis/converters/RDKit.py

```diff
@@ -435,9 +440,16 @@ def atomgroup_to_mol(ag, NoImplicit=True, max_iter=200, force=False):
         # infer bond orders and formal charges from the connectivity
         _infer_bo_and_charges(mol)
         mol = _standardize_patterns(mol, max_iter)
+        # reorder atoms to match MDAnalysis
+        order = np.argsort([atom.GetIntProp("_MDAnalysis_index")
+                            for atom in mol.GetAtoms()])
+        mol = Chem.RenumberAtoms(mol, order.astype(int).tolist())
```

Comment:
Question: Is this necessary? Shouldn't _Mdanalysis_index have been set sequentially above in the `enumerate` call?

Reply:
Also, `MDAnalysis_index` is a bit ambiguous in the sense that I don't really know if it's the `atom.index` property or index in that particular atom group. Possibly out of scope for this PR.

Reply:
Oh wait, I'm guessing this is because the RDKit reactions will have messed up the order. Would you mind adding a comment to that effect?